plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java

```diff
@@ -54,6 +54,68 @@
     private IDocument document;
     private IPath location;
 
+    private static class LineCol {
+        @Override
+        public int hashCode() {
+            final int prime = 31;
+            int result = 1;
+            result = prime * result + col;
+            result = prime * result + line;
+            return result;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (!(obj instanceof LineCol)) {
+                return false;
+            }
+            LineCol other = (LineCol) obj;
+            if (col != other.col) {
+                return false;
+            }
+            if (line != other.line) {
+                return false;
+            }
+            return true;
+        }
+
+        private final int line;
+        private final int col;
+
+        public LineCol(int line, int col) {
+            super();
+            this.line = line;
+            this.col = col;
+        }
+    }
+
+    private static class MessageInfo {
+        private String message;
+        private final int markerSeverity;
+        private final String messageId;
+        private final int line;
+        private final int column;
+        private final String docLineContents;
+
+        public MessageInfo(String message, int markerSeverity, String messageId, int line, int column,
+                String docLineContents) {
+            super();
+            this.message = message;
+            this.markerSeverity = markerSeverity;
+            this.messageId = messageId;
+            this.line = line;
+            this.column = column;
+            this.docLineContents = docLineContents;
+        }
+
+        public void addMessageLine(String message) {
+            this.message += "\n" + message;
```

Comment:
Since it's a message to be mutated, I suggest you change the message to be a `FastStringBuffer` and then append the newline and then append the message (doing string concatenation this way requires reallocating the whole memory at each new step and thus isn't ideal).